_episodes/script.md

```diff
@@ -1,38 +1,223 @@
+- "How to capture output of a script?"
 objectives:
-- "See episode body and [this issue](https://github.com/common-workflow-lab/cwl-novice-tutorial/issues/8)"
+- "Include and run a script in a step at runtime"
+- "Capture output of a script"
 keypoints:
 - "First key point. Brief Answer to questions. (FIXME)"
 ---
+
 By the end of this episode,
 learners should be able to
-__include their own script as a step in a workflow__
-
-several options for this were identified during the lesson development sprint.
-we should choose one or two to focus on in the tutorial
-(add your thoughts at the corresponding issue [here](https://github.com/common-workflow-lab/cwl-novice-tutorial/issues/8)):
-  - make script executable and add to path
-      - advantage: quick
-      - downside:
-          - the scripts are not shipped with the CWL document itself
-          - not portable to other execution environment without putting the script first
-  - pack the scripts into a docker container and make them executable there:
-      - advantage: portable
-      - downside:
-          - only works for people that are using containers
-  - use InitialWorkDirRequirement to list script content directly in the CWLToolWrapper
-      - example: https://github.com/CompEpigen/ATACseq_workflows/blob/master/CWL/tools/generate_atac_signal_tags.cwl (bad example as here the scripts are too long, only for short scripts)
-      - advantage:
-          - portable
-          - can be used independent of the dependency management solution
-      - disadvantage:
-          - explodes the complexity of a CWL tool wrapper
-          - having to deal with escapes and so on
-  - distribute as seperate package via pip / cran / ...
+__include and run their own script in a step at runtime__
+
+> ## Exercise 1:
+>
+> Complete the `requirements` field for the `CommandLineTool` that creates a script at runtime. As a HINT, `DockerRequirement` has already been filled-in.
+>
+> ~~~
+> class: CommandLineTool
+> cwlVersion: v1.1
+> requirements:
+>   DockerRequirement:
+>     dockerPull: 'debian:stable'
+>   -----------------:
+> ~~~
+> {: .language-yaml }
+>
+> > ## Solution
+> >
+> > ~~~
+> > class: CommandLineTool
+> > cwlVersion: v1.1
+> > requirements:
+> >    DockerRequirement:
+> >      dockerPull: 'debian:stable'
+> >    InitialWorkDirRequirement:
+> > ~~~
+> > {: .language-yaml }
+> >
+> {: .solution}
+{: .challenge}
+
+> ## Exercise 2:
+>
+> Identify `CWL` keywords for defining script name `script.sh` and the contents of this script under `InitialWorkDirRequirement`.
+>
+> ~~~
+> InitialWorkDirRequirement:
+>   listing:
+>     - ------: script.sh
+>       ------: |
+>         echo "*Documenting input*" && \
+>         echo "Input received: $(inputs.message)" && \
+>         echo "Exit"
+>
+> inputs:
+>   message:
+>     type: string
+> ~~~
+> {: .language-yaml }
+>
+> > ## Solution:
+> >
+> > ~~~
+> > class: CommandLineTool
+> > cwlVersion: v1.1
+> > requirements:
+> >    DockerRequirement:
```

Comment:
These lines are not present in the code block for the exercise, above. Should they be removed here? Or added in the previous block?

Reply:
I think it's better to remove these. As these are not required as part of the exercise concept.

Reply:
👍 If you remove those four lines I think we can get this merged.

Reply:
Done under this [commit](https://github.com/common-workflow-lab/cwl-novice-tutorial/pull/20/commits/d9e3fe71304b11480f9c4dd48458818aeea2147f)